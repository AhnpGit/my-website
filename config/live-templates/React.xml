<templateSet group="React">
  <template name="rx_module" value="const _ACTION_PREFIX = '$projectName$/$moduleName$'&#10;&#10;const $ACTION_NAME$ = `${_ACTION_PREFIX}/$ACTION_NAME$`&#10;&#10;// TODO: Connect the reducer.&#10;export default function reducer(state = {$defaultState$}, action) {&#10;    switch (action.type) {&#10;        case $ACTION_NAME$:&#10;            return {$END$}&#10;&#10;        default:&#10;            return state&#10;    }&#10;}&#10;&#10;export const $actionCreator$ = () =&gt; ({&#10;    type: $ACTION_NAME$,&#10;    payload: {$payload$},&#10;    meta: {},&#10;    error: undefined,&#10;})&#10;" description="Creates a DUCKS-module" toReformat="true" toShortenFQNames="true">
    <variable name="projectName" expression="" defaultValue="&quot;&lt;project name&gt;&quot;" alwaysStopAt="true" />
    <variable name="moduleName" expression="fileNameWithoutExtension()" defaultValue="" alwaysStopAt="true" />
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <variable name="defaultState" expression="" defaultValue="" alwaysStopAt="true" />
    <variable name="actionCreator" expression="" defaultValue="&quot;myActionCreator&quot;" alwaysStopAt="true" />
    <variable name="payload" expression="" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_component" value="import React from 'react'&#10;&#10;const $ComponentName$ = ({$props$}) =&gt; (&#10;    &lt;$END$/&gt;&#10;)&#10;&#10;export default $ComponentName$" description="Creates a functional component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <variable name="props" expression="" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_component_class" value="import React from 'react'&#10;&#10;class $ComponentName$ extends React.Component {&#10;    render() {&#10;        return (&#10;            &lt;$END$/&gt;&#10;        )&#10;    }&#10;}&#10;&#10;export default $ComponentName$" description="Creates a class component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_proptypes" value="import PropTypes from 'prop-types'&#10;$ComponentName$.propTypes = {&#10;    $prop$: PropTypes.$END$,&#10;}" description="Adds proptypes to a component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <variable name="prop" expression="" defaultValue="&quot;prop&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_action_async" value="const $ACTION_NAME$ = `${_ACTION_PREFIX}/$ACTION_NAME$`&#10;const $ACTION_NAME$_PENDING = `${$ACTION_NAME$}_PENDING`&#10;const $ACTION_NAME$_FULFILLED = `${$ACTION_NAME$}_FULFILLED`&#10;const $ACTION_NAME$_REJECTED = `${$ACTION_NAME$}_REJECTED`" description="Creates async action types" toReformat="true" toShortenFQNames="true">
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_action" value="const $ACTION_NAME$ = `${_ACTION_PREFIX}/$ACTION_NAME$`" description="Creates an action type" toReformat="true" toShortenFQNames="true">
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_creator" value="export const $actionCreator$ = () =&gt; ({&#10;    type: $ACTION_NAME$,&#10;    payload: {$END$},&#10;    meta: {},&#10;    error: undefined,&#10;})" description="Creates an action creator" toReformat="true" toShortenFQNames="true">
    <variable name="actionCreator" expression="" defaultValue="&quot;myActionCreator&quot;" alwaysStopAt="true" />
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_creator_async" value="export const $actionCreator$ = () =&gt; ({&#10;    type: $ACTION_NAME$,&#10;    payload: getPayload(),&#10;    meta: {},&#10;    error: undefined,&#10;    &#10;    async function getPayload() {&#10;        return $END$&#10;    }&#10;})" description="Creates an async action creator" toReformat="true" toShortenFQNames="true">
    <variable name="actionCreator" expression="" defaultValue="&quot;myActionCreator&quot;" alwaysStopAt="true" />
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_container" value="import { connect } from 'react-redux'&#10;import $ComponentName$ from '../components/$ComponentName$'&#10;&#10;const mapStateToProps = (state) =&gt; ({&#10;    $END$&#10;})&#10;&#10;const mapDispatchToProps = (dispatch) =&gt; ({&#10;})&#10;&#10;export default connect(&#10;    mapStateToProps,&#10;    mapDispatchToProps,&#10;)($ComponentName$)" description="Creates a container class" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_styled" value="import styled from 'styled-components'&#10;&#10;const $ComponentName$ = styled.$ElementType$`&#10;  $END$&#10;`&#10;&#10;export default $ComponentName$" description="Creates a styled component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <variable name="ElementType" expression="" defaultValue="&quot;div&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
</templateSet>
