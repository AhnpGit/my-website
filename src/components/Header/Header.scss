header {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  z-index: map_get($layers, fixed-floating);
  background-color: $trout;
}

.wrapper__header {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;

  height: 60px;
  color: $white;

  @include step-scale('padding-left', $base-spacer, $base-unit, $base-unit);
  @include step-scale('padding-right', $base-spacer, $base-unit, $base-unit);

  @include media-query($medium-up) {
    height: 80px;
  }
}

.header__logo {
  img {
    height: 50px;

    @include media-query($medium-up) {
      height: 70px;
    }
  }
}

.nav__links {
  display: flex;
}

.compactmenu__link {
  @include on-event {
    border: none;
  }
}

.nav__link {
  text-transform: uppercase;
  font-size: 90%;
  text-decoration: none;

  @include on-event {
    border-bottom: none;
  }

  @include media-query($medium-up) {
    color: $mischka;
    border-bottom: 2px solid transparent;
    transition: border-bottom 0.2s;

    @include on-event {
      color: $white;
      border-bottom: 2px solid $tahitiGold;
    }
  }
}

.nav__link:not(:last-child) {
  margin-right: $base-unit;
}

.modal__body {
  ul {
    li {
      line-height: 39px;
      border-bottom: 1px solid $athens-gray;
      padding: 0 $base-unit;
    }
  }
}
