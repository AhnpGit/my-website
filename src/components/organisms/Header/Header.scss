.header__wrapper {
  @include clearFix();
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  z-index: map_get($zIndex, fixedFloating);
}

header {
  position: relative;
  width: 100%;
  z-index: map_get($zIndex, hoverActiveFocus);

  .row {
    margin-bottom: 0;
  }
}
