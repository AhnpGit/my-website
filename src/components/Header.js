import React from "react";

const Header = () => (
    <header>
        <nav>
            <div className="row">
                <img src={require(`../assets/img/logo-white.png`)} alt="Omnifood logo" className="logo" />
                <img src={require(`../assets/img/logo.png`)} alt="Omnifood logo" className="logo-black" />
                <ul className="main-nav">
                    <li><a href="#features">Food delivery</a></li>
                    <li><a href="#works">How it works</a></li>
                    <li><a href="#cities">Our cities</a></li>
                    <li><a href="#plans">Sign up</a></li>
                </ul>
                <a className="mobile-nav-icon"><i className="ion-navicon-round"></i></a>
            </div>
        </nav>
        <div className="hero-text-box">
            <h1>Goodbye junk food.<br />Hello super healthy meals.</h1>
            <a className="btn btn-full" href="">I'm hungry</a>
            <a className="btn btn-ghost" href="">Show me more</a>
        </div>
    </header>
)

export default Header