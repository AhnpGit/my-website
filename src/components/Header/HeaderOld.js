// @flow

import { clearFix, rem } from 'polished'
import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'
import CategoryNavigation from '../../../containers/CategoryNavigation'
import CategoryNavigationAndSearch from '../../../containers/CategoryNavigationAndSearch'
import SearchForm from '../../../containers/SearchForm'
import type { DeviceInfo } from '../../modules/deviceInfo'
import type { ShopData } from '../../../modules/shop'
import { base } from '../../../utils/style/branding'
import colors from '../../../utils/style/colors'
import { media, stepScale } from '../../../utils/style/mixins'
import { zIndex } from '../../../utils/style/settings'
import Row from '../../elements/Row/Row'
import Sprite from '../../atoms/Sprite/Sprite'
import Wrapper from '../../atoms/Wrapper/Wrapper'
import MainNavigation from '../../molecules/MainNavigation/MainNavigation'

type Arguments = {
  position?: 'fixed' | 'static',
  navigationMenuIsOpen?: boolean,
  deviceInfo: DeviceInfo,
  shop: ShopData
}

/**
 * Main Header, used on all 'pages' Contains the logo, the main search bar and housed the main navigation
 * The logo is embedded, the search and main navigation are taken in as children
 *
 * The three children are split off to be able to test them separately and in order to be able to make those components reusable
 */

const Header = ({ position = 'fixed', navigationMenuIsOpen = false, deviceInfo, shop }: Arguments) => {

  return (
    <ComponentWrapper position={position}>
      <Component>
        <StyledRow>
          <HeaderTop>
            <TopWrapper>
              <HeaderLogo>
                <LogoSprite name="KrampLogo" />
              </HeaderLogo>
              {!deviceInfo.deviceType.isSmall ? (
                <HeaderSearch>
                  <SearchForm />
                </HeaderSearch>
              ) : ''}
              <HeaderMainNavigation>
                <MainNavigation shop={shop} />
              </HeaderMainNavigation>
            </TopWrapper>
          </HeaderTop>
        </StyledRow>
        <StyledRow>
          <HeaderBottom>
            <BottomWrapper>
              {deviceInfo.deviceType.isSmall ? <CategoryNavigationAndSearch /> : <CategoryNavigation />}
            </BottomWrapper>
          </HeaderBottom>
        </StyledRow>
      </Component>

      {navigationMenuIsOpen && <NavigationMenuBackdrop />}
    </ComponentWrapper>
  )
}

export const ComponentWrapper = styled.div`
  ${ clearFix() };
  position: ${({ position }: { position: 'fixed' | 'static' }) => position};
  top: 0;
  left: 0;
  width: 100%;
  z-index: ${ zIndex.fixedFloating };
`

const Component = styled.div`
  position: relative;
  width: 100%;
  z-index: ${ zIndex.hoverActiveFocus };
`
const StyledRow = styled(Row)`
  margin-bottom: 0;
`

const NavigationMenuBackdrop = styled.div`
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background: rgba(0, 0, 0, 0.3);
  z-index: ${ zIndex.aboveGround };
`
NavigationMenuBackdrop.displayName = 'NavigationMenuBackdrop'

const HeaderTop = styled.div`
  height: ${rem('50px')};
  background-color: ${colors.brightRed};
  color: ${ colors.white };

  ${ media.mediumUp`
    height: ${ rem('70px')};
  ` }
`

const TopWrapper = styled(Wrapper)`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  ${stepScale('padding-left', rem('8px'), base.unit, base.unit)};
  ${stepScale('padding-right', rem('8px'), base.unit, base.unit)}
  height: 100%;
  
  .ie9 & {
    display: block;
  }
`

const LogoSprite = styled(Sprite)`
  width: ${rem('119px')};
  height: ${rem('18px')};

  ${ media.mediumUp `
    width: ${rem('173px')};
    height: ${rem('26px')};
  `}
  
  .ie9 & {
    display: inline-block;
    margin-top: ${ rem('22px') };
  }
`

const HeaderBottom = styled.div`
  background-color: ${colors.redBerry};
  color: ${colors.white};
  height: ${ rem('50px')};

  ${ media.mediumUp`
    height: auto;
  ` }
`

const BottomWrapper = styled(Wrapper)`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  height: 100%;
  padding: 0 ${rem('5px')};

  ${ media.mediumUp`
      padding: 0;
      justify-content: flex-start;
  ` }
  
  .ie9 & {
    display: block;
  }
`

const HeaderLogo = styled.div`
  display: flex;
  
  .ie9 & {
    width: 33%;
    float: left;
  }
  
  .ie10 & {
    flex: 1 1 auto;
  }
`

const HeaderSearch = styled.div`
  width: 100%;

  ${ media.mediumUp`
    width: ${ rem('250px')};
  ` }
  
  .ie9 & {
    float: left;
    
    ${ media.mediumUp`
      width: 33%;
    ` }
  }
  
  .ie10 & {
    flex: 0 0 250px;
  }
`

const HeaderMainNavigation = styled.div`
  .ie9 & {
    float: right;
  }
  
  .ie10 & {
    flex: 1 1 auto;
    justify-content: flex-end;
  }
`

Header.propTypes = {
  /** dynamic position, used for styleguide only */
  position: PropTypes.string,

  /** Whether or not the navigation menu is currently open. Used to show a backdrop. */
  navigationMenuIsOpen: PropTypes.bool,

  /**
   * Object contains the data of the shop
   */
  shop: PropTypes.object.isRequired
}

export default Header
