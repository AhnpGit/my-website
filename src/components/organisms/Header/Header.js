// @flow

import React from "react"
import PropTypes from 'prop-types'
import Icon from '../../atoms/Icon/Icon'
import type { DeviceInfo } from '../../../utils/types'

type HeaderArguments = {
  deviceInfo: DeviceInfo,
  isOpen: boolean,
  setOpenState: (isOpen: boolean) => mixed
}

/*
 * Header component displays a logo on the left and navigation bar on the right
 */
const Header = ({ deviceInfo, isOpen, setOpenState } : HeaderArguments) => {
  return (
    <header>
      <nav>
        <div className="logo">
          <img src={require('../../../assets/img/logo-white.png')} alt="Omnifood logo" className="logo" />
        </div>
        {deviceInfo.deviceType.isSmall
          ? renderCompactMenu(isOpen, setOpenState)
          : renderNormalNavigation()}
      </nav>
    </header>
  )
}

function renderCompactMenu(isOpen: boolean, setOpenState: (isOpen: boolean) => mixed) {
  return (
    <div className="main__nav__compactmenu">
      <a href="" className="compactmenu__link" onClick={(event) => {
        event.preventDefault()
        setOpenState(!isOpen)
      }}>
        <Icon name="category" />
      </a>
    </div>
  )
}

function renderNormalNavigation() {
  return (
    <ul className="main-nav">
      <li className="nav__item"><a className="main-nav__link" href="#features">Food delivery</a></li>
      <li className="nav__item"><a className="main-nav__link" href="#works">How it works</a></li>
      <li className="nav__item"><a className="main-nav__link" href="#cities">Our cities</a></li>
      <li className="nav__item"><a className="main-nav__link" href="#plans">Sign up</a></li>
    </ul>
  )
}

Header.propTypes = {
  /**
   * Info about the device used here to determine screen size
   */
  deviceInfo: PropTypes.object.isRequired,

  /**
   * Boolean indicates the open state of the navigation menu on small screen
   */
  isOpen: PropTypes.bool.isRequired,

  /**
   * Function which triggers open or close navigation menu on small screen
   */
  setOpenState: PropTypes.func.isRequired
}

export default Header
