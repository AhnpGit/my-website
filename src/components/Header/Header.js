// @flow

import React from "react"
import PropTypes from 'prop-types'

import Icon from '../../elements/Icon/Icon'
import type { DeviceInfo } from '../../utils/types'

import './Header.scss'
import Modal from '../Modal/Modal'
import ModalHeader from '../Modal/ModalHeader'

import '../Modal/ModalHeader.scss'

type HeaderArguments = {
  deviceInfo: DeviceInfo,
  isOpen: boolean,
  setOpenState: (isOpen: boolean) => mixed
}

/*
 * Header component displays a logo on the left and navigation bar on the right
 */
const Header = ({ deviceInfo, isOpen, setOpenState }: HeaderArguments) => {
  return (
    <div className="header__wrapper">
      <header>
        <div className="header__top">
          <div className="top__wrapper">
            <div className="header__logo">
              <img src={require('../../assets/img/logo-white.png')} alt="Omnifood logo" className="logo" />
            </div>
            {!deviceInfo.deviceType.isSmall
              ? renderNormalNavigation()
              : renderCompactMenu(isOpen, setOpenState)
            }
          </div>
        </div>
      </header>
    </div>
  )
}

function renderCompactMenu(isOpen: boolean, setOpenState: (isOpen: boolean) => mixed) {
  return (
    <nav className="nav__compactmenu">
      <a href="" className="compactmenu__link" onClick={(event) => {
        event.preventDefault()
        setOpenState(!isOpen)
      }}>
        <Icon name="category" />
      </a>
      <Modal
        isOpen={isOpen}
        contentLabel="CategoryNavigationModal Menu"
        onRequestClose={() => setOpenState(false)}
        displayMode="smallScreen"
      >
        <ModalHeader
          onRequestClose={() => setOpenState(false)}
          closeButtonTitle="Close modal"
        >
          Menu
        </ModalHeader>
        <div className="modal__body">
          <ul>
            <li><a className="nav__link" href="#features">Food delivery</a></li>
            <li><a className="nav__link" href="#works">How it works</a></li>
            <li><a className="nav__link" href="#cities">Our cities</a></li>
            <li><a className="nav__link" href="#plans">Sign up</a></li>
          </ul>
        </div>
      </Modal>
    </nav>
  )
}

function renderNormalNavigation() {
  return (
    <nav className="nav__links">
      <a className="nav__link" href="#features">Food delivery</a>
      <a className="nav__link" href="#works">How it works</a>
      <a className="nav__link" href="#cities">Our cities</a>
      <a className="nav__link" href="#plans">Sign up</a>
    </nav>
  )
}

Header.propTypes = {
  /**
   * Info about the device used here to determine screen size
   */
  deviceInfo: PropTypes.object.isRequired,

  /**
   * Boolean indicates the open state of the navigation menu on small screen
   */
  isOpen: PropTypes.bool.isRequired,

  /**
   * Function which triggers open or close navigation menu on small screen
   */
  setOpenState: PropTypes.func.isRequired
}

export default Header
