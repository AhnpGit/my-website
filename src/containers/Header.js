import { connect } from 'react-redux'
import Header from '../components/Header/Header'
import { setOpenState } from '../modules/header'
import type { DeviceInfo, Dispatcher } from '../utils/types'

type HeaderType = {
  deviceInfo: DeviceInfo,
  isOpen: boolean
}

export const mapStateToProps = (state : HeaderType) => ({
  deviceInfo: state.deviceInfo,
  isOpen: state.header.isOpen
})

export const mapDispatchToProps = (dispatch : Dispatcher) => ({
  setOpenState: (isOpen) => {
    dispatch(setOpenState(isOpen))
  }
})

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Header)
