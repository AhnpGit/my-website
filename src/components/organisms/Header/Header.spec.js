import React from 'react'
import renderer from 'react-test-renderer'
import Header from './Header'

describe('Header components', () => {
  it('renders a logo and a navigation button on small screen', () => {
    const deviceInfo = {
      deviceType: {
        isSmall: true
      }
    }

    const result = renderer.create(
      <Header deviceInfo={deviceInfo} />
    ).toJSON()

    expect(result).toMatchSnapshot()
  })

  it('renders a logo and normal navigation on medium up screen', () => {
    const deviceInfo = {
      deviceType: {
        isSmall: false
      }
    }

    const result = renderer.create(
      <Header deviceInfo={deviceInfo} />
    ).toJSON()

    expect(result).toMatchSnapshot()
  })
})
