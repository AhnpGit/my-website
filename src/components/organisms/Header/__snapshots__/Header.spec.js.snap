// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header components renders a logo and a navigation button on small screen 1`] = `
<header>
  <nav>
    <div
      className="logo"
    >
      <img
        alt="Omnifood logo"
        className="logo"
        src="logo-white.png"
      />
    </div>
    <div
      className="main__nav__compactmenu"
    >
      <a
        className="compactmenu__link"
        href=""
        onClick={[Function]}
      >
        <svg
          aria-labelledby="icon-category-title"
          className={undefined}
          role="img"
        >
          <title>
            category
          </title>
          <use
            pointerEvents="none"
            xlinkHref="#icons-category"
          />
        </svg>
      </a>
    </div>
  </nav>
</header>
`;

exports[`Header components renders a logo and normal navigation on medium up screen 1`] = `
<header>
  <nav>
    <div
      className="logo"
    >
      <img
        alt="Omnifood logo"
        className="logo"
        src="logo-white.png"
      />
    </div>
    <ul
      className="main-nav"
    >
      <li
        className="nav__item"
      >
        <a
          className="main-nav__link"
          href="#features"
        >
          Food delivery
        </a>
      </li>
      <li
        className="nav__item"
      >
        <a
          className="main-nav__link"
          href="#works"
        >
          How it works
        </a>
      </li>
      <li
        className="nav__item"
      >
        <a
          className="main-nav__link"
          href="#cities"
        >
          Our cities
        </a>
      </li>
      <li
        className="nav__item"
      >
        <a
          className="main-nav__link"
          href="#plans"
        >
          Sign up
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
